src/main/java/org/kuali/kfs/kim/document/IdentityManagementPersonDocument.java

```diff
@@ -482,8 +486,20 @@ public void doRouteStatusChange(final DocumentRouteStatusChange statusChangeEven
         if (getDocumentHeader().getWorkflowDocument().isProcessed()) {
             setIfGroupsEditable();
             setIfRolesEditable();
+
+            for (final PersonDocumentRole role : roles) {
+                for (final KimDocumentRoleMember rolePrncpl : role.getRolePrncpls()) {
+                    if (ObjectUtils.isNull(rolePrncpl.getActiveFromDate())) {
```

Comment:
I am not sure the best approach just yet, but I think there might be an attribute of rolePrncpl that might work for us.